src/Nethermind/Nethermind.Evm/TransactionProcessing/SystemTransactionProcessor.cs

```diff
@@ -0,0 +1,64 @@
+// SPDX-FileCopyrightText: 2024 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using Nethermind.Core;
+using Nethermind.Core.Specs;
+using Nethermind.Evm.Tracing;
+using Nethermind.Int256;
+using Nethermind.Logging;
+using Nethermind.Specs;
+using Nethermind.State;
+
+namespace Nethermind.Evm.TransactionProcessing;
+
+public sealed class SystemTransactionProcessor : TransactionProcessorBase
+{
+    private readonly bool _isAura;
+
+    /// <summary>
+    /// Hacky flag to execution options, to pass information how original validate should behave.
+    /// Needed to decide if we need to subtract transaction value.
+    /// </summary>
+    private const int OriginalValidate = 2 << 30;
+
+    public SystemTransactionProcessor(ISpecProvider? specProvider,
+        IWorldState? worldState,
+        IVirtualMachine? virtualMachine,
+        ICodeInfoRepository? codeInfoRepository,
+        ILogManager? logManager)
+        : base(specProvider, worldState, virtualMachine, codeInfoRepository, logManager)
+    {
+        _isAura = SpecProvider.SealEngine == SealEngineType.AuRa;
+    }
+
+    protected override TransactionResult Execute(Transaction tx, in BlockExecutionContext blCtx, ITxTracer tracer, ExecutionOptions opts) =>
+        base.Execute(tx, in blCtx, tracer, !opts.HasFlag(ExecutionOptions.NoValidation)
+            ? opts | (ExecutionOptions)OriginalValidate | ExecutionOptions.NoValidation
```

Comment:
We force NoValidation mode here, and save original value under `OriginalValidate` flag for estimate gas edge case.